overlord/devicestate/devicemgr.go

```diff
@@ -329,22 +333,79 @@ func (m *DeviceManager) maybeSetupUbuntuSave() error {
 	if !runMntSaveMounted {
 		// we don't have ubuntu-save, save will be used directly
 		logger.Noticef("no ubuntu-save mount")
-		m.saveAvailable = true
 		return nil
 	}
 
-	logger.Noticef("bind-mounting ubuntu-save under %v", dirs.SnapSaveDir)
+	sysd := systemd.New(systemd.SystemMode, progress.Null)
+
+	// In newer core20/core22 we have a mount unit for ubuntu-save, which we
+	// will try to start first. Invoking systemd-mount in this case would fail.
+	err = sysd.Start([]string{"var-lib-snapd-save.mount"})
+	if err == nil {
+		logger.Noticef("mount unit for ubuntu-save was started")
+		return nil
+	} else {
+		// We only fall through and mount directly if the failure was because of a missing
+		// mount file, which possible does not exist. Any other failure we treat as an actual
+		// error.
+		if !strings.Contains(err.Error(), "Unit var-lib-snapd-save.mount not found.") {
```

Comment:
this feels a bit fragile, but I also know we tried to do better and had to fight a bit differences between systemd versions. We should add at least something like:

// XXX systemd ideally should start returning some kind UnitNotFound errors in this situation

Reply:
Yea you are absolutely right - good point about adding the comment, I did this. Thank you for the input, and thanks for the review!